src/state/opamGlobalState.ml

```diff
@@ -60,7 +60,12 @@ let load lock_kind =
     OpamConsole.error_and_exit `Configuration_error
       "Opam has not been initialised, please run `opam init'";
   let config_lock = OpamFilename.flock lock_kind (OpamPath.config_lock root) in
-  let config = load_config lock_kind global_lock root in
+  let config =
+    try load_config lock_kind global_lock root
+    with OpamFormatUpgrade.Upgrade_done _ as e ->
+      OpamSystem.funlock config_lock;
```

Comment:
Wondering if we should put this lock release in `OpamCliMain`, when `Upgrade_done` is catched?

Reply:
It doesn’t have the right config lock to close at that point?l, I think?